validators/cgroup_validator.go

```diff
@@ -82,6 +103,21 @@ func (c *CgroupsValidator) validateCgroupSubsystems(cgroups, subsystems []string
 }
 
 func (c *CgroupsValidator) getCgroupSubsystems() ([]string, error) {
+	// /proc/cgroups is meaningless for v2
+	// https://github.com/torvalds/linux/blob/v5.3/Documentation/admin-guide/cgroup-v2.rst#deprecated-v1-core-features
+	if IsCgroup2UnifiedMode() {
+		// "pseudo" controllers do not appear in /sys/fs/cgroup/cgroup.controllers.
+		// - devices: implemented in kernel 4.15
+		// - freezer: implemented in kernel 5.2
+		// We assume these are always available, as it is hard to detect availability.
+		pseudo := []string{"devices", "freezer"}
+		data, err := ioutil.ReadFile("/sys/fs/cgroup/cgroup.controllers")
+		if err != nil {
+			return nil, err
+		}
+		subsystems := append(pseudo, strings.Fields(string(data))...)
```

Comment:
i guess we'd need cgroup_validator_test.go unit tests for this.

Reply:
@neolit123 That is reasonable but it is difficult to test the current implementation of getCgroupSubsystems(). Should we refactor it or do you have any ideas?

Reply:
current unit tests are only for validateCgroupSubsystems(). the hardcoded / preudo controllers should be unit tested at least.

Reply:
I added unit tests for validateCgroupSubsystems() to test when hardcoded subsystems values("devices" and "freezer") are set.